draft/0004-user-sign-in-experience.md

```diff
@@ -0,0 +1,669 @@
+
+## 8.Future possibilities
+
+### 8.1.User profile management
+
+The verification and user profile APIs can be extended to support more advanced user profile management features, outside the sign-in experience. For example, users can update their profile data, manage their MFA settings, and view their interaction history using the experience APIs through a dedicated user profile management interface.
+
+### 8.2 Simplified experience API calls
+
+Reduce the complexity of the experience API calls. We are considering merging several API calls into a single API call to simplify the user interaction flow. For example, omit the extra submit request for the sign-in experience, consider introducing a `autoSubmit` flag in some of the interaction APIs to automatically submit the interaction once all the requirements are met.
+
+### 8.3 Provide additional custom user profiles in a sign-in experience
+
+The user profile data can be extended to support custom user profile fields in the sign-in experience. Developers can define and set required custom profile fields for the user during the interaction. This will allow developers to collect additional user data and customize the user profile based on their specific use cases.
+
+### 8.4 Experience API SDK
+
+Develop a client-side SDK that simplifies the integration of the experience APIs into the front-end application. The SDK can provide a set of pre-built components and functions that handle the interaction flow, user profile management, and MFA settings. This will help developers quickly integrate the experience APIs into their applications and build a customized sign-in experience with minimal effort.
+
+### 8.5 Additional password and social/SSO verification APIs
+
+Currently the password verification record was created directly via the sign-in user identification API. We may provide additional verification APIs to create and verify the password verification record separately. This will allow developers to customize the password verification process and create a password verification record outside the sign-in experience.
+Similarly, we can provide additional verification APIs to create and verify the social/SSO verification record separately. This will allow developers to customize the social/SSO authentication process and create a social/SSO verification record outside the sign-in experience as well.
+
+### 8.6 Open verification APIs
+
+The verification APIs as well as the verification record can be extended to support other verification circumstances outside the sign-in experience. For example, support dynamic MFA verification and the verification for the user profile updates.
```

Comment:
```suggestion
### 8.5 Standalone verification APIs

Currently, a verification record is created directly via the Experience API. We may provide standalone verification APIs to create and verify these verification records without an interaction context. This will allow developers to customize the verification process and use it in a separate flow, e.g., building a GitHub-like sudo mode.
```